rust/uefi/stub/src/thin.rs

```diff
@@ -125,19 +124,15 @@ pub fn boot_linux(handle: Handle, dynamic_initrds: Vec<Vec<u8>>) -> uefi::Result
     // that are supposedly measured in TPM2.
     // Therefore, it is normal to not verify their hashes against a configuration.
 
-    /// Compute the necessary padding based on the provided length
-    fn compute_pad4(len: usize) -> Vec<u8> {
-        vec![0u8; (4 - (len % 4)) % 4]
-    }
-
-    initrd_data.append(&mut compute_pad4(initrd_data.len()));
+    // Pad to align
+    initrd_data.resize(initrd_data.len().next_multiple_of(4), 0);
```

Comment:
I don't think this is correct. Previously we essentially doubled the size of `initrd_data` by appending a vec of zeros of it's own length. Now, we resize it to it's own length. Which basically only aligns the size to a multiple of four but otherwise is  a no-op.